fix meta for fft_c2c

garfield1997 opened the pull request:
Fixes #ISSUE_NUMBER
The original meta logic for cuda was incorrect when the input dimension was greater than cufft_max_ndim.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

garfield1997 committed:
fix meta for fft_c2c

shunting314 reviewed (commented):
I don't think I'm the right reviewer for this PR since I've never touched pytorch fft codepath before. cc @ezyang / @eellison

garfield1997 commented on torch/_meta_registrations.py:
```diff
@@ -316,11 +319,23 @@ def meta_fft_c2c(self, dim, normalization, forward):
+            strides = working_tensor.stride()
+            sorted_dims.sort(
+                key=lambda i: strides[i], reverse=True
+            )  # NB reverse!  Not sure if this is og bug
```
The `_sort_dims` function appears to be intended for CPU branches, where the use of stride differs from that in CUDA branches, so it might not be suitable for use here.

garfield1997 commented:
I updated the code based on Claude's suggestion and switched to using non-continuous input for testing. Could you please take a look if you have time? @ezyang

garfield1997 commented:
If you have time, please help look at this pr, thanks. @ezyang

jansel commented:
@pytorchbot rebase

jansel reviewed (changes requested):
fix merge conflicts

garfield1997 commented:
The related changes are already addressed by PR https://github.com/pytorch/pytorch/pull/177323, so I am closing this one.
